### PR TITLE
cp2k 2022.1

### PR DESCRIPTION
**Pull request by cho-m**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

**Comment by carlocab** [comment]
```
  gfortran -D__FFTW3 -I/home/linuxbrew/.linuxbrew/opt/fftw/include -I /home/linuxbrew/.linuxbrew/opt/libxc/include -I /home/linuxbrew/.linuxbrew/Cellar/cp2k/2022.1/libexec/include -fopenmp -funroll-loops -ftree-vectorize -ffree-form -O2 -g -D__FFTW3 -D__NO_STATM_ACCESS -D__ACCELERATE -D__LIBXC -D__LIBINT -D__COMPILE_ARCH="\"Darwin-gfortran\"" -D__COMPILE_DATE="\"Sun Aug  7 02:33:37 UTC 2022\"" -D__COMPILE_HOST="\"fbf7241727f9\"" -D__COMPILE_REVISION="\"git:2caeb6f\"" -D__DATA_DIR="\"/tmp/cp2k-20220807-11090-c2phcx/cp2k-2022.1/data\""  -L/tmp/cp2k-20220807-11090-c2phcx/cp2k-2022.1/lib/Darwin-gfortran/ssmp -o /tmp/cp2k-20220807-11090-c2phcx/cp2k-2022.1/exe/Darwin-gfortran/memory_utilities_unittest.ssmp memory_utilities_unittest.o  -lcp2kcommon -lcp2koffload -lcp2kmpiwrap -lcp2kbase -L/tmp/cp2k-20220807-11090-c2phcx/cp2k-2022.1/lib/Darwin-gfortran/ssmp/exts/dbcsr -ldbcsr -L/home/linuxbrew/.linuxbrew/opt/fftw/lib -lfftw3 -lfftw3_threads -lstdc++ -framework Accelerate -L/home/linuxbrew/.linuxbrew/opt/libxc/lib -lxcf03 -lxc -L/home/linuxbrew/.linuxbrew/Cellar/cp2k/2022.1/libexec/lib -lint2
  gfortran: error: unrecognized command-line option '-framework'
```

**Comment by carlocab** [comment]
#105024 might help.

File: Formula/cp2k.rb [hunk 1]
```diff
@@ -1,10 +1,9 @@
 class Cp2k < Formula
   desc "Quantum chemistry and solid state physics software package"
   homepage "https://www.cp2k.org/"
-  url "https://github.com/cp2k/cp2k/releases/download/v8.2.0/cp2k-8.2.tar.bz2"
-  sha256 "2e24768720efed1a5a4a58e83e2aca502cd8b95544c21695eb0de71ed652f20a"
+  url "https://github.com/cp2k/cp2k/releases/download/v2022.1/cp2k-2022.1.tar.bz2"
```
Comment (carlocab) [inline Formula/cp2k.rb:4]: We should double-check that upstream have changed their version scheme permanently. It'll be very annoying to change it back if they decide to go back to the old one.

Comment (cho-m) [reply Formula/cp2k.rb:4]: It looks like it based on resources/comments available.

Last meeting notes https://www.cp2k.org/_media/dev:cp2k_dev_meeting_05_05_22.pdf
> - switching to year.month or year.number naming schema?
> - yes, (year).(release number in year)
 
Google groups release https://groups.google.com/g/cp2k/c/CjSsK2BX8LQ/m/ne2ypAu-AAAJ

And comment in another project https://github.com/aiidateam/aiida-cp2k/issues/163#issuecomment-1181747449
> The pace of development in CP2K has notably picked up over the last years. Therefore we now adopted a 6 month release cycle. Alongside, we changed our versioning scheme from `<major>.<minor>` to `<year>.<number>` to make it more self-explanatory.

**Comment by cho-m** [comment]
> #105024 might help.

Upstream has dropped most Linux gfortran files. The only remaining one is OpenCL psmp variant.

There is another Linux-gnu-x86_64.psmp variant which may be usable if we set the Makefile variables to disable some libraries (e.g. `"USE_COSMA=", "USE_ELPA="`) and override path to Homebrew's libraries. Not sure yet.

There is no usable Linux ssmp variant given that the existing ones are for static libraries.

Alternative would be to try to use upstream scripts (e.g. https://github.com/cp2k/cp2k/blob/master/tools/toolchain/install_cp2k_toolchain.sh) which is how they create Docker images. It will require a lot on input args since most defaults are set to build particular versions of dependencies (e.g. `"--with-openblas=system", "--with-elpa=no", ...). Not sure how easy it is to use.

File: Formula/cp2k.rb [hunk 2]
```diff
@@ -35,47 +38,95 @@ def install
...
+        --with-cosma=no
+        --with-libvori=no
+      ]
+      args << "--generic" if build.bottle?
+
+      cd "tools/toolchain" do
+        # Need OpenBLAS source to get proc arch info in scripts/get_openblas_arch.sh
+        Formula["openblas"].stable.stage Pathname.pwd/"build/OpenBLAS"
+
+        system "./install_cp2k_toolchain.sh", *args
```
Comment (cho-m) [inline Formula/cp2k.rb:113]: May need to check if need to add wrapper to binaries given script outputs:
```
  +----------------------------------------------------------------+
  |  YOU SHOULD ALWAYS SOURCE ./install/setup BEFORE YOU RUN CP2K  |
  |  COMPILED WITH THIS TOOLCHAIN                                  |
  +----------------------------------------------------------------+
```

File: Formula/cp2k.rb [hunk 3]
```diff
@@ -35,47 +38,95 @@ def install
...
+        --with-gsl=no
+        --with-libvdwxc=no
+        --with-spglib=no
+        --with-hdf5=no
+        --with-spfft=no
+        --with-spla=no
+        --with-cosma=no
+        --with-libvori=no
+      ]
+      args << "--generic" if build.bottle?
```
Comment (cho-m) [inline Formula/cp2k.rb:107]: Probably not necessary. Default is `-mtune=native` but superenv probably drops it. Same thing for `--generic` arg which sets `-mtune=generic`.
```suggestion
```

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**